postinstall.jl on Mac OS X

running `include("postinstall.jl")` on Mac OS X gives:

```
uname: illegal option -- o
usage: uname [-amnprsv]
uname: illegal option -- o
usage: uname [-amnprsv]
gcc -Wall -I /Users/bjohnson/src/julia/usr/src -O2 -shared -fPIC profile.c   -o libprofile.so
profile.c:1:28: error: support/dtypes.h: No such file or directory
profile.c:4: error: expected ‘)’ before ‘*’ token
profile.c:6: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘*’ token
profile.c: In function ‘sprofile_bt’:
profile.c:27: warning: implicit declaration of function ‘rec_backtrace’
profile.c:27: error: ‘bt_data_prof’ undeclared (first use in this function)
profile.c:27: error: (Each undeclared identifier is reported only once
profile.c:27: error: for each function it appears in.)
profile.c: In function ‘sprofile_init’:
profile.c:130: error: ‘bt_data_prof’ undeclared (first use in this function)
profile.c:132: error: ‘ptrint_t’ undeclared (first use in this function)
profile.c:132: error: expected expression before ‘)’ token
profile.c: In function ‘sprofile_get_data’:
profile.c:140: error: ‘bt_data_prof’ undeclared (first use in this function)
make: *** [profile] Error 1
failed process: Process(`make INCPATH=/Users/bjohnson/src/julia/usr/src`, ProcessExited(2)) [2]
 in pipeline_error at process.jl:385
 in run at process.jl:374
 in include at boot.jl:248
 in include_from_node1 at util.jl:251
at /Users/bjohnson/.julia/Profile/src/postinstall.jl:5
```

Changing the Makefile to use clang rather than gcc does not appear to help anything.

--- a/deps/Makefile
+++ b/deps/Makefile
@@ -1,6 +1,6 @@
 # Build with something like:
 #   make INCPATH=~/src/julia/src
-# Calling require("postinstall.jl") from within Julia should handle
+# Calling require("build.jl") or Pkg.runbuildscript("Profile") from within Julia should handle
 # this for you.
 
 CC=gcc
@@ -9,19 +9,25 @@ CFLAGS=-Wall -I $(INCPATH)
 LDFLAGS=
 LINKLIBS=
 
-platform=$(shell uname -o)
-ifeq ($(platform),GNU/Linux)
+platform=$(shell uname)
+ifeq ($(platform),Linux)
 	LINKLIBS+=-lrt
-endif
-ifeq ($(platform),Darwin)
+	LIBEXT=so
+else ifeq ($(platform),Darwin)
 	LDFLAGS+=-Wl,-U,_rec_backtrace
+	LIBEXT=dylib
+else
+	# WINDOWS
+	# what are the appropriate LDFLAGS??
+	LDFLAGS+=-lrt
+	LIBEXT=dll
 endif
 
 profile: profile.c
-	$(CC) $(CFLAGS) -O2 -shared -fPIC profile.c $(LDFLAGS) $(LINKLIBS) -o libprofile.so
+	$(CC) $(CFLAGS) -O2 -shared -fPIC profile.c $(LDFLAGS) $(LINKLIBS) -o libprofile.$(LIBEXT)
 
 clean:
-	-rm libprofile.so
+	-rm libprofile.$(LIBEXT)
 
 .PHONY: \
 	profile

